helix-core/src/main/java/org/apache/helix/controller/dataproviders/BaseControllerDataProvider.java

```diff
@@ -89,7 +89,7 @@
   // Property caches
   private final PropertyCache<ResourceConfig> _resourceConfigCache;
   private final PropertyCache<InstanceConfig> _instanceConfigCache;
-  private final PropertyCache<LiveInstance> _liveInstanceCache;
+  protected final PropertyCache<LiveInstance> _liveInstanceCache;
```

Comment:
nit, safer to add a protected readonly method for getting the list instead of returning the list. Otherwise, the children may modify this list.